operations/host_spawn.go

```diff
@@ -681,6 +702,49 @@ func hostRunCommand() cli.Command {
 			client := conf.setupRestCommunicator(ctx)
 			defer client.Close()
 
+			var hostIDs []string
+			if hostID != "" {
+				hostIDs = []string{hostID}
+			} else {
+				var createdBeforeTime, createdAfterTime time.Time
+				if createdBefore != "" {
+					createdBeforeTime, err = time.Parse(time.RFC3339, createdBefore)
+					if err != nil {
+						return errors.Wrap(err, "can't parse created before time")
+					}
+				}
+				if createdAfter != "" {
+					createdAfterTime, err = time.Parse(time.RFC3339, createdAfter)
+					if err != nil {
+						return errors.Wrap(err, "can't parse create after time")
+					}
+				}
+
+				hosts, err := client.GetHosts(ctx, model.APIHostParams{
```

Comment:
Can this get hosts that are not owned by this user?

Reply:
If the user it not a superuser we set Mine true [on the server side](https://github.com/evergreen-ci/evergreen/pull/3042/files#diff-8872b9ded06791dd850f93fdbc7018dfR365).